### PR TITLE
chore: Implement grouping wallets by the network

### PR DESCRIPTION
**Pull request by ignromanov**
## Summary

- Add new components to group wallets by the network
- Update dashboard
- Add special page to see all the network wallets

## Checklist

-   [x] My changes look good in both light AND dark mode
-   [x] The change is not hardcoded to a single network, but has multi-asset in mind
-   [x] I checked my changes for obvious issues, debug statements and commented code
-   [x] Documentation _(if necessary)_
-   [x] Tests _(if necessary)_
-   [x] Ready to be merged

### CODE REVIEW

**Comment by faustbrian** [comment]
Please link the GitHub issues you are resolving so they are automatically closed when this is merged.

https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue

**Comment by anil203** [comment]
1. When the “Send” button is disabled for a wallet, the corresponding overflow menu icon is also displayed as disabled (grayed out). The overflow menu icon has to be displayed in active state (Green/Blue).
image

2. The overflow menu should contain only two options – “Rename” and “Delete”, as per spec. As per the current implementation it shows many other options.
image

3. Logo of the devnet/testnet assets should be shown in Gray.
image

4. The names of devnet/testnet assets should be followed by devnet symbol: **</>**
image

5. The “Total Currency” field for devnet assets should be displayed as “NA”
image

6. On hover of any asset tile, there has to be a shadow displayed as mentioned in the spec doc and designs. The current implementation does not show any such shadow.
image

7. As per the designs: When there are no wallets available then the message to be displayed is “Your portfolio is currently empty. Create or Import a wallet to get started.”. As per current implementation, the message displayed is “You don't have any wallets yet.”. Need to update the same.
image

8. Incorrect line thickness and color https://tppr.me/d7DwL In the design, the line thickness is 2px and the color is green-100 https://tppr.me/tEjga.

9. There is hover state missing when hovering over the wallet group https://tppr.me/URYmL. There should be a hover as in the design https://tppr.me/eI1Jw .

10. Padding between line and amount should be 16px https://tppr.me/500Kb. The indent may vary, depending on the amounts in the cards https://tppr.me/OwsgP , for example what is highlighted in red always has the same, and what is highlighted in orange varies.

11. There should be padding from the edge to the background of the hover of 8px https://tppr.me/ZpQJ0, as in the design of https://tppr.me/bxYxy.

12. If you have the latest transactions activated they don't show in the Portfolio https://tppr.me/t478v.

13. Starred addresses should always be at the top of the list https://tppr.me/vnqpc

14. The data must be aligned so that | is the same spot table https://tppr.me/glWjK, should follow design [](https://tppr.me/w2w3N).

**Review by dated — COMMENTED** [review]
- the overflow of the tables should be 8px inside the portfolio tables, as opposed to the regular 16px (so there is a gap between the border and the hovered table):

image

- entering the wallet menu with the mouse triggers some unexpected hover effects on the table and the icons in the menu:

image


- Wallets / Total Balance / Total Currency should be aligned

image

File: src/app/components/WalletListItem/WalletListItem.blocks.tsx [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+					"ring-theme-background": !isSelected,
+					"ring-theme-success-100": isSelected && !shouldUseDarkColors(),
+					"ring-theme-success-900": isSelected && shouldUseDarkColors(),
+				},
+			),
+		[isSelected],
+	);
+
+	return (
+		<TableCell variant="start" innerClassName="space-x-4" isCompact={isCompact}>
```
Comment (dated) [inline src/app/components/WalletListItem/WalletListItem.blocks.tsx:58]:
```suggestion
		<TableCell variant="start" innerClassName={cn(isCompact ? "space-x-3" : "space-x-4")} isCompact={isCompact}>
```

File: src/app/components/WalletListItem/WalletListItem.blocks.tsx [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+			<div
+				className={cn(
+					"flex justify-center gap-4 divide-x divide-theme-secondary-300 dark:divide-theme-secondary-800",
+					{
+						"h-11": !isCompact,
+						"h-5": isCompact,
+					},
```
Comment (dated) [inline src/app/components/WalletListItem/WalletListItem.blocks.tsx:65]:
```suggestion
					"flex justify-center divide-x divide-theme-secondary-300 dark:divide-theme-secondary-800",
					{
						"h-11 gap-4": !isCompact,
						"h-5 gap-3": isCompact,
					},
```

File: src/app/components/WalletListItem/WalletListItemSkeleton.tsx [hunk 3]
```diff
@@ -1,41 +1,79 @@
...
 	return (
 		<TableRow data-testid="WalletListItemSkeleton">
 			<TableCell variant="start" innerClassName="space-x-4" isCompact={isCompact}>
+				<div
+					className={cn(
+						"flex justify-center gap-4 divide-x divide-theme-secondary-300 dark:divide-theme-secondary-800",
+						{
+							"h-11": !isCompact,
+							"h-5": isCompact,
+						},
```
Comment (dated) [inline src/app/components/WalletListItem/WalletListItemSkeleton.tsx:24]:
```suggestion
						"flex justify-center divide-x divide-theme-secondary-300 dark:divide-theme-secondary-800",
						{
							"h-11 gap-4": !isCompact,
							"h-5 gap-3": isCompact,
						},
```

File: src/app/components/WalletListItem/WalletListItemSkeleton.tsx [hunk 4]
```diff
@@ -1,41 +1,79 @@
...
-}
-
-export const WalletListItemSkeleton = ({ isCompact }: WalletListItemSkeletonProperties) => {
-	const circleSize = isCompact ? 20 : 44;
+export const WalletListItemSkeleton: React.VFC<WalletListItemSkeletonProperties> = ({ isCompact }) => {
+	const { t } = useTranslation();
 
 	return (
 		<TableRow data-testid="WalletListItemSkeleton">
 			<TableCell variant="start" innerClassName="space-x-4" isCompact={isCompact}>
```
Comment (dated) [inline src/app/components/WalletListItem/WalletListItemSkeleton.tsx:17]:
```suggestion
			<TableCell variant="start" innerClassName={cn(isCompact ? "space-x-3" : "space-x-4")} isCompact={isCompact}>
```

File: src/domains/dashboard/components/NetworkWallets/NetworkWalletsList.tsx [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+				Header: t("COMMON.CURRENCY"),
+				// eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+				accessor: (wallet) => wallet.convertedBalance?.(),
+				cellWidth: "w-32",
+				className: "justify-end",
+			},
+			{
+				Header: " ",
+				cellWidth: "w-32",
+				className: "justify-end",
```
Comment (dated) [inline src/domains/dashboard/components/NetworkWallets/NetworkWalletsList.tsx:48]:
```suggestion
				Header: "Actions",
				cellWidth: "w-32",
				className: "hidden no-border",
```

File: src/app/components/WalletListItem/WalletListItemSkeleton.tsx (hunk 4 above)
Comment (ignromanov) [reply src/app/components/WalletListItem/WalletListItemSkeleton.tsx:17]: Ahh, they have different gaps and spaces for the compact and the regular version. Hadn't noticed, thanks!

**Comment by anil203** [comment]
1.	The total balance is overlapping with the vertical bar | when there is lengthy value displayed.
image

2.	From the overflow menu -> Select “Wallet Name” -> Update the wallet name -> Save the changes -> The updated name does not reflect immediately in the wallet column. (Refer to the attached video for better understanding. Video timestamp 0:00 - 0:30)

3.	In dark mode, the hover state of wallet group is different as per the designs.  
image

4.	Empty portfolio message is displayed twice when there are no wallets added to the profile.  
image

5.	When there is any starred wallet present in testnet asset group, the testnet wallet group is displayed first and then the corresponding public network wallet group. (Refer to the attached video for better understanding. Video timestamp 0:35 till the end)

Video for point #2 and #5:

https://user-images.githubusercontent.com/10047242/144403882-ba7e0ce4-01eb-46b6-8fd1-9efa5a5585b4.mp4

**Comment by anil203** [comment]
Hey @ignromanov, Just one thing which broke (Probably due to the fix for Point #4 in the [PR comments](https://github.com/PayvoHQ/wallet/pull/606#issuecomment-984493571)):

In the portfolio breakdown section, when the user does not select any public network from the filters section, then the app throws a message indicating the same. As shown in the image below.

Probably while fixing the issue #4 mentioned in [PR comments](https://github.com/PayvoHQ/wallet/pull/606#issuecomment-984493571), this message was removed. Alessandro can provide you more details in case you need any.

Once this is resolved, I guess we are good to go.

Screenshot 2021-12-03 at 12 47 34 PM

**Comment by samharperpittam** [comment]
@ignromanov - Is it possible to stop the highlighting of the next token name when opening and closing the dropdown?

See Blockpool in the video 

https://user-images.githubusercontent.com/44671015/144853931-41d00f32-d707-4039-b15c-6d321961da02.mov

**Comment by ignromanov** [comment]
> @ignromanov - Is it possible to stop the highlighting of the next token name when opening and closing the dropdown?
fixed

**Review by anil203 — DISMISSED** [review]

**Pull request merged**